ocp_resources/data_source.py

```diff
@@ -65,3 +66,23 @@ def pvc(self):
                 f"dataSource {self.name} is pointing to a non-existing PVC, name: {pvc_name}, "
                 f"namespace: {pvc_namespace}"
             )
+
+    def wait_for_status(self, status, timeout=TIMEOUT_4MINUTES, sleep=1):
```

Comment:
why can't ocp_resources.resource.Resource.wait_for_condition be used?

Reply:
you should enhance ocp_resources.resource.Resource.wait_for_condition
add optional reason and if passed, match is as well